faunadb-java-dsl/src/main/java/com/faunadb/client/types/Enums.java

```diff
@@ -0,0 +1,52 @@
+package com.faunadb.client.types;
+
+import com.faunadb.client.types.Value.StringV;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static com.google.common.collect.Maps.newEnumMap;
+import static com.google.common.collect.Maps.newHashMap;
+
+final class Enums {
+  private Enums() {}
+
+  private final static ConcurrentHashMap<Class<?>, Map<Enum, StringV>> ENCODING_MAP = new ConcurrentHashMap<>();
+  private final static ConcurrentHashMap<Class<?>, Map<String, Enum>> DECODING_MAP = new ConcurrentHashMap<>();
+
+  public static Map<Enum, StringV> getEncodingMap(Class<Enum> enumClass) {
+    Map<Enum, StringV> enumMap = ENCODING_MAP.get(enumClass);
+
+    if (enumMap == null) {
+      enumMap = newEnumMap(enumClass);
+
+      for (Enum en : enumClass.getEnumConstants()) {
+        try {
+          java.lang.reflect.Field field = enumClass.getField(en.name());
+          FaunaEnum faunaEnum = field.getAnnotation(FaunaEnum.class);
+          String enumName = faunaEnum != null ? faunaEnum.value() : en.name();
+          enumMap.put(en, new StringV(enumName));
+        } catch (NoSuchFieldException ex) {
```

Comment:
this will never happen, since it's iterating over the enum constants, I just had to add try/catch because it's a checked exception